crates/gossip/src/respond/accept.rs

```diff
@@ -90,12 +90,13 @@ impl K2Gossip {
 
         match accept.snapshot {
             Some(their_snapshot) => {
+                let their_snapshot: DhtSnapshot = their_snapshot.into();
                 let (next_action, _) = self
                     .dht
                     .read()
                     .await
                     .handle_snapshot(
-                        their_snapshot.into(),
+                        their_snapshot.clone(),
```

Comment:
Cloning, which I'd prefer not to do with this type but we need to compare the snapshot before updating arcs and we can't pass a reference across the box future boundary here